jcdb-core/src/main/kotlin/org/utbot/jcdb/impl/types/signature/JvmTypes.kt

```diff
@@ -16,54 +16,103 @@
 
 package org.utbot.jcdb.impl.types.signature
 
+import kotlinx.metadata.KmType
 import org.utbot.jcdb.api.PredefinedPrimitives
+import org.utbot.jcdb.api.ext.isNullable
 
-sealed class JvmType {
+sealed class JvmType(val isNullable: Boolean) {
 
     abstract val displayName: String
 
+    abstract fun setNullability(nullable: Boolean): JvmType
+
+    open fun relaxWithKmType(kmType: KmType): JvmType =
```

Comment:
For sure this is shouldn't be interface method of JvmType

Reply:
Rewrote this logic using extension functions